### DISCUSSION TITLE
Convert gif to png sprite.

### DISCUSSION THREAD

**Discussion by Tyme-Bleyaert** [saucecontrol/PhotoSauce#106] [Q&A] [answered]
Can I use this library the same way I can use SkiaSharp?

I currently use SkiaSharp to convert a gif to a sprite as followed:

```
using (var tempSurface = SKSurface.Create(new SKImageInfo(width, height)))
{
    //get the drawing canvas of the surface
    var canvas = tempSurface.Canvas;

    //set background color
    canvas.Clear(SKColors.Transparent);

    //go through each image and draw it on the final image
    int offset = 0;
    int offsetTop = 0;
    int count = 0;
    foreach (SKBitmap image in images)
    {
        canvas.DrawBitmap(image, SKRect.Create(offset, offsetTop, image.Width, image.Height));
        count++;

        offset = (image.Width * (count % columns)) + padding * (count % columns);
        if (count % columns == 0)
            offsetTop += image.Height + padding;

    }

    // return the surface as a manageable image
    finalImage = tempSurface.Snapshot();
}
```
But SkiaSharp doesn't have a Gif encoder. So I'm using this library to scale gif's.

I was wondering if I could replace SkiaSharp entirely.

**Comment by saucecontrol** [comment, marked as answer]
It would be possible, but it would be very inefficient with the API as currently designed because you'd have to start the decode process over for each extracted frame.

In general, I don't have plans to expand the API to cover all general-purpose image manipulation (SkiaSharp or ImageSharp will be your best bets for that), but this is kind of an interesting case for a custom image encoder because you're outputting them stacked/wrapped vertically so they wouldn't have to be in memory all at once.  I'll add this to my list of considerations for when I open up those APIs to public.
